### ISSUE TITLE
feature: Add --force option to `yarn link` to overwrite existing link

### ISSUE THREAD

**Issue by dnaf** [yarnpkg/yarn#7216]
Currently, trying to `yarn link` a package which is already registered spits out a warning and does nothing. Manually removing the existing link adds unnecessary tedium. Having a `yarn link --force` flag which replaces the existing link would be very useful.

Related issue: #7054

**Comment by zwhitchcox** [comment]
What is the process to unlink a directory that has been deleted? Do we have to create a whole new one just to delete it? I agree this would be a much better way.

**Comment by zwhitchcox** [comment]
At least tell us where the folder that is supposedly linked is, so we know where to delete

**Comment by MichalBryxi** [comment]
Bumped into the same. The case is quite simple: I'm doing some work on project 1, other work on project 2, other work on project 3. And suddenly:

```
❯ yarn link
yarn link v1.17.3
warning There's already a package called "my-addon" registered. This command has had no effect. If this command was run in another folder with the same name, the other folder is still linked. Please run yarn unlink in the other folder if you want to register this folder.
```

But ... what is the **other folder**?  Where did it come from? I assume that yarn _knows_ the location of the _other folder_ so it would be more helpful to tell me: 

> There's already a package called "my-addon" registered in `${my-addon-path}`. This command has had no effect. If this command was run in another folder with the same name, the other folder is still linked. Please run `cd ${my-addon-path}; yarn unlink` before you try to register this folder.

**Comment by dominicfallows** [comment]
I created https://www.npmjs.com/package/manage-linked-packages to help with this, whilst Yarn doesn't have a `--force` option

**Comment by phoenixeliot** [comment]
AFAICT `yarn unlink` will unlink based on the package name in package.json, rather than based on the folder name. Is that correct?
> If this command was run in another folder with the same name, the other folder is still linked.

I think the "folder with the same name" part is misleading, given that. In my testing it doesn't matter what the folder is named, nor do you need to find the "other folder" and unlink from there; any folder with the same name in its package.json should work.

**Comment by ChuckJonas** [comment]
This feature is desperately needed.  Or at least a `yarn links` command that would list out all the links with their paths.  I apparently linked a package in the wrong place and it's hard to find

**Comment by sujanp-kr** [comment]
clear entry in: ~/.config/yarn/link/

**Comment by phoenixeliot** [comment]
I think `(yarn unlink || true) && yarn link` is equivalent to `yarn unlink; yarn link`

**Comment by k80w** [comment]
@phoenixeliot This is true so long as the shell isn't set to exit on a command failure (i.e. when the `-e` flag is set). It is perhaps good practice to explicitly allow `yarn unlink` to fail regardless.

Note that the subshell in that command isn't necessary; `yarn unlink || true && yarn link` or `yarn unlink || true; yarn link` behave almost identically to that example.

**Comment by nerdess** [comment]
What is the status of this?

I am currently manually deleting links I do not need anymore in the _./config/yarn/link_ folder. This is my clumsy "unlink" process.

<img width="804" alt="Bildschirmfoto 2023-11-23 um 15 34 18" src="https://github.com/yarnpkg/yarn/assets/711925/0457e463-b55b-4dcc-9b82-511a072619d4">
